server/lib/routes/secret.js

```diff
@@ -0,0 +1,72 @@
+'use strict'
+
+async function deleteTokenIfAuthorized(
+  firebaseAdmin,
+  tokenId,
+  subscriptionId,
+  user
+) {
+  const db = firebaseAdmin.firestore()
+  const subscriptionRef = await db
+    .collection('subscriptions')
+    .where(firebaseAdmin.firestore.FieldPath.documentId(), '==', subscriptionId)
+    .where('userId', '==', user)
+    .get()
+
+  if (subscriptionRef.empty) {
+    return false
+  }
+
+  await db
+    .collection('allTokens')
+    .doc(tokenId)
+    .delete()
+
+  return true
```

Comment:
I'm not sure this is secure. Doesn't this mean that as long as we find a record in the subscriptions collection with the provided subscription id and user id, a user can delete ANY token?

Reply:
The subscription id associated with that token particular token is passed in so if the subscription id of that particular token is not associated with the user it won't progress to deletion.